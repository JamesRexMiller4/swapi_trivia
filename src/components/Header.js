import React from 'react';
import './Header.scss';

const Header = ({name, quote, ranking}) => {
  return (
    <header className="header">
      <h2></h2>

    </header>
  )
}

export default Header;