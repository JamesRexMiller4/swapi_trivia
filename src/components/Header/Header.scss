@import '../variables.scss';

.header {
  background-color: $main-background-color;
  display: flex;
  height: 20vh;
  justify-content: space-around;
  width: 100%;
}

.logo-container, .info-container {
  width: 40%;
}

.logo-container {
  margin-left: -8%;
  text-align: center;
}

.logo-h1-header {
  font-size: 4.5rem;
  margin-bottom: 5%;
}

.quote-h3-header {
  color: $primary-accent-color;
  font: {
    family: $small-text-font;
    size: 1.6rem;
  }
}

.info-container {
  color: $primary-accent-color;
  font-family: $small-text-font;
  max-height: 100%;
  padding-top: 0.5%;
  text-align: right;
}

.welcome-message {
  font-size: 2rem;
}

.ranking-h3-header {
  font-size: 1.7rem;
}

.ranking {
  display: inline;
  font-size: 1.4rem;
  margin-left: 1%;
}

.logout-button-header {
  background-color: $secondary-background-color;
  font-family: $small-text-font;
  height: 30px;
  margin-top: 2%;
  outline: none;
  width: 15%;
  &:active {
    background-color: $main-header-color;
  }
}
