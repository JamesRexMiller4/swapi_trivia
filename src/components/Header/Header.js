import React from 'react';
import PropTypes from 'prop-types';
import './Header.scss';

const Header = ({name, quote, ranking, logout}) => {
  return (
    <header className="header">
      <div className="logo-container">
        <h1 className="logo-h1-header">Swapi WarS</h1>
        <h3 className="quote-h3-header">"{quote}"</h3>
      </div>
      <div className="info-container">
        <h2 className="welcome-message">Welcome, {name}!</h2>
        <h3 className="ranking-h3-header">Ranking: <span className="ranking">{ranking}</span></h3>
        <button className="logout-button-header" id="logout-btn" onClick={logout}>Log Out</button>
      </div>
    </header>
  )
}

Header.PropTypes = {
  name: PropTypes.string,
  quote: PropTypes.string,
  ranking: PropTypes.string,
  logout: PropTypes.func
}

export default Header;
