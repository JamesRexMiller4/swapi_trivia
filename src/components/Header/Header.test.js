import React from 'react';
import ReactDOM from 'react-dom';
import Header from './Header';
import {shallow} from 'enzyme';

describe('Header', () => {
  let wrapper, mockLogOut, user;

  beforeEach(() => {
    mockLogOut = jest.fn()

    wrapper = shallow(<Header 
      name={'Leeroy Jenkings'}
      quote={'LEEERRRROOOOYYY JEEEENNNNKKKKIIINNNSSS'}
      ranking={'expert'}
      logout={mockLogOut}
    />)
  })

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Header />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('should match the snapshot with all the data passed through', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should call logout when a user clicks the logout button', () => {
    wrapper.find('#logout-btn').simulate('click');

    expect(mockLogOut).toHaveBeenCalled();
  })
});