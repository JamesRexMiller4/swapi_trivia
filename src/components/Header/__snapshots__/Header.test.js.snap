// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match the snapshot with all the data passed through 1`] = `
<header
  className="header"
>
  <div
    className="logo-container"
  >
    <h1
      className="logo-h1-header"
    >
      Swapi WarS
    </h1>
    <h3
      className="quote-h3-header"
    >
      "
      LEEERRRROOOOYYY JEEEENNNNKKKKIIINNNSSS
      "
    </h3>
  </div>
  <div
    className="info-container"
  >
    <h2
      className="welcome-message"
    >
      Welcome, 
      Leeroy Jenkings
      !
    </h2>
    <h3
      className="ranking-h3-header"
    >
      Ranking: 
      <span
        className="ranking"
      >
        expert
      </span>
    </h3>
    <button
      className="button-header"
      id="logout-btn"
      onClick={[MockFunction]}
    >
      Log Out
    </button>
    <Link
      key="favs"
      to="/favorites"
    >
      <button
        className="button-header"
        id="toggle-fav-btn"
      >
        View Favs
      </button>
    </Link>
  </div>
</header>
`;
